specs/phase0/validator.md

```diff
@@ -328,6 +328,12 @@ def get_eth1_vote(state: BeaconState, eth1_chain: Sequence[Eth1Block]) -> Eth1Da
     valid_votes = [vote for vote in state.eth1_data_votes if vote in votes_to_consider]
 
     # Default vote on latest eth1 block data in the period range unless eth1 chain is not live
+    # FIXME: Non-substantive casting: a workaround for passing typing linter for future patch forks
+    state_eth1_data = Eth1Data(
+        deposit_root=state.eth1_data.deposit_root,
+        deposit_count=state.eth1_data.deposit_count,
+        block_hash=state.eth1_data.block_hash,
+    )
```

Comment:
/cc @protolambda 

This is a sequela of OOP SSZ of the new `remerkleable`. The linter doesn't recognize the Phase0 fields in the new Lightclient `BeaconState` statically. They are `object` type to mypy.

A quick workaround here is to define a `state_eth1_data = Eth1Data(...)` variable since `max()` function requires homogeneous list.

1. Do you think it's possible to fix it in `remerkleable`?
2. Or, is there a better way to casting `Container`?

If not, not ideal but I'm okay to leave it as-is for now and open an issue... 😅

Reply:
1. No, it's non-standard already. We can do anything with python, but can't make mypy interpret it all I'm afraid. I'm not sure how we could make it recognize the inherited fields correctly. Suggestions/PRs to remerkleable are welcome though.
2. I dislike explicit casting explicitly by wrapping with a constructor, since it comes at a performance cost. Maybe a typed variable definition like `foobar: MyTypeHere = foobar_obj_something` works to narrow down the type?

Reply:
`foobar: MyTypeHere = foobar_obj_something` works! 👍 👍